### PR TITLE
OSDOCS-10430: Create ROSA HCP distro

### PR DESCRIPTION
**Pull request by bmcelvee**
Version(s):
4.15+ 

Issue:
https://issues.redhat.com/browse/OSDOCS-10430

Link to docs preview:

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: _distro_map.yml [hunk 1]
```diff
@@ -187,6 +187,19 @@ openshift-rosa:
     rosa-preview:
```
Comment (bmcelvee) [inline _distro_map.yml:187]: Would it be too much effort to change the dir to `rosa-classic`?

File: _distro_map.yml [hunk 2]
```diff
@@ -187,6 +187,19 @@ openshift-rosa:
...
+openshift-rosa-hcp:
+  name: Red Hat OpenShift Service on AWS
+  author: OpenShift Documentation Project <openshift-docs@redhat.com>
+  site: commercial
+  site_name: Documentation
+  site_url: https://docs.openshift.com/
+  branches:
+    enterprise-4.15:
+      name: ''
+      dir: rosa-hcp/
```
Comment (bmcelvee) [inline _distro_map.yml:199]: If we change the ROSA classic dir, then we could make this `rosa` instead of `rosa-hcp`

File: _distro_map.yml (hunk 1 above)
Comment (laubai) [reply _distro_map.yml:187]: docs.os folks please do correct me, but I'm pretty sure this defines the directory name to save the built content in, and uses that same dir name as part of the URL for docs.openshift.com, right?

So we should be able to define one of these for rosa-classic and one for rosa, and just ensure we're using the right build conditions for each distro, correct?

Comment (bmcelvee) [reply _distro_map.yml:187]: It sounds like team consensus is that we're okay with keeping the ROSA with classic architecture dir  name as `rosa` and the new ROSA distro having the dir name `rosa-hcp`.

File: _distro_map.yml (hunk 2 above)
Comment (kalexand-rh) [inline _distro_map.yml:199]: This distro should be password-protected whenever you merge this PR. Please ping me when you do so I can make sure that everything is correct when you merge it.

Comment (bmcelvee) [reply _distro_map.yml:199]: Thank you!

**Pull request merged**